packages/ckeditor5-table/tests/tablecolumnresize/tablecolumnresizeediting.js

```diff
@@ -1946,6 +1946,104 @@ describe( 'TableColumnResizeEditing', () => {
 		} );
 	} );
 
+	describe( 'Hide/show resizers', () => {
+		let model, editor, view, editorElement;
+
+		beforeEach( async () => {
+			editorElement = document.createElement( 'div' );
+			document.body.appendChild( editorElement );
+			editor = await createEditor();
+
+			model = editor.model;
+			view = editor.editing.view;
+		} );
+
+		afterEach( async () => {
+			if ( editorElement ) {
+				editorElement.remove();
+			}
+
+			if ( editor ) {
+				await editor.destroy();
+			}
+		} );
+
+		describe( 'on mousedown/mouseup', () => {
+			it( 'should not hide resizers when resizer is clicked', () => {
+				setModelData( model,
+					'<table columnWidths="100%">' +
+					'<tableRow>' +
+					'<tableCell>' +
+					'<paragraph>' +
+					'[foo]' +
+					'</paragraph>' +
+					'</tableCell>' +
+					'</tableRow>' +
+					'</table>'
+				);
+
+				const resizer = view.getDomRoot().querySelector( '.table-column-resizer' );
+
+				resizer.dispatchEvent( new MouseEvent( 'mousedown', { bubbles: true } ) );
+
+				const viewRoot = view.domConverter.mapDomToView( view.getDomRoot() );
+
+				expect( viewRoot.hasClass( 'ck-resizers-hidden' ) ).to.be.false;
+			} );
+
+			it( 'should hide resizers when table cell is clicked and show resizers on mouseup', () => {
+				setModelData( model,
+					'<table columnWidths="100%">' +
+					'<tableRow>' +
+					'<tableCell>' +
+					'<paragraph>' +
+					'[foo]' +
+					'</paragraph>' +
+					'</tableCell>' +
+					'</tableRow>' +
+					'</table>'
+				);
+
+				const td = view.getDomRoot().querySelector( 'td' );
+
+				td.dispatchEvent( new MouseEvent( 'mousedown', { bubbles: true } ) );
+
+				const viewRoot = view.domConverter.mapDomToView( view.getDomRoot() );
+
+				expect( viewRoot.hasClass( 'ck-resizers-hidden' ) ).to.be.true;
+
+				td.dispatchEvent( new MouseEvent( 'mouseup', { bubbles: true } ) );
+
+				expect( viewRoot.hasClass( 'ck-resizers-hidden' ) ).to.be.false;
+			} );
+		} );
+
+		describe( 'on resizer mouse drag', () => {
+			it( 'does not change table selection', () => {
+				setModelData( model, modelTable( [
+					[ '00[]', { contents: '01', colspan: 2 } ],
+					[ '10', '11', '12' ]
+				], { columnWidths: '50%,25%,25%' } ) );
+
+				const paragraph = view.domConverter.mapViewToDom(
+					view.document.getRoot().getChild( 0 ).getChild( 1 ).getChild( 1 ).getChild( 1 ).getChild( 2 ).getChild( 0 )
```

Comment:
This chain doesn't look healthy :smile: We could improve it in various ways:

1.  Split the chain links to separate lines with comments which element it is, like:

```
view.document.getRoot() // div
	.getChild( 0 ) // figure
	.getChild( 1 ) // table
	(...)
```

2\. Get the dom root right away and use the `querySelector()`.

The same goes for the resizer const.